# Duplicate entries in a row


In [21]:
import pandas as pd

In [22]:
df = pd.read_csv('sloganlist.csv')

In [23]:
df.head()


Company                          Slogan
0  Costa Coffee              For coffee lovers.
1         Evian              Evian. Live young.
2        Dasani  Designed to make a difference.
3      Heineken        It's all about the beer.
4      Gatorade           The Legend Continues.

In [24]:
# shape of the dataset
df.shape

(1162, 2)

In [25]:
# columns of the dataset
df.columns

Index(['Company', 'Slogan'], dtype='object')

In [26]:
# Duplicate entries in the specific column
print(df.Company.duplicated().sum(), df.Slogan.duplicated().sum())


593 594


In [27]:
# Duplicate entries in the form of rows
df.duplicated().sum()

593

Hence there are 593 rows which are duplicated in the dataset. These are:

# Seeing the duplicated rows



In [28]:
# keeping first duplicates
df.loc[df.duplicated(keep='first'), :]

Company                            Slogan
26             Heineken          It's all about the beer.
27             Gatorade             The Legend Continues.
60              Colgate         Stronger, healthier gums.
61                Fanta      Drink Fanta, stay Bamboocha.
62            Pizza Hut                    Make it great.
...                 ...                               ...
1157       Levi's Jeans  Original jeans. original people.
1158  Dove (toiletries)     The secret of beautiful hair.
1159             Lipton                      Be more tea.
1160            Pampers     Discover your baby’s world.
1161          Coca-Cola                Taste The Feeling.

[593 rows x 2 columns]

In [29]:
# keeping first duplicates
df.loc[df.duplicated(keep='last'), :]

Company                         Slogan
3                        Heineken       It's all about the beer.
4                        Gatorade          The Legend Continues.
20                        Colgate      Stronger, healthier gums.
21                          Fanta   Drink Fanta, stay Bamboocha.
22                      Pizza Hut                 Make it great.
...                           ...                            ...
1124                      Pampers  Discover your baby’s world.
1125                    Coca-Cola             Taste The Feeling.
1133               Domino's Pizza         Pizza deliver experts.
1134                    Pizza Hut                 Make it great.
1140  KFC(Kentucky Fried Chicken)           Finger Lickin' Good.

[593 rows x 2 columns]

# Dropping duplicates from the dataframe


In [30]:
# Remaining dataset
df.drop_duplicates(keep='first').shape

(569, 2)

This is a simple example of how we can deal with the duplicates. But in a real life dataset dealing with machine learning algorithms we can have a very large number of independent variables. Some are relevant and some are not. In that case we can check for the relevant independent variables, check for duplicated in those columns only and remove the duplicated according to that.

We should also heck the distribution of the classes with duplicates and without. It can be that a positive class can have many duplicates and than you have imbalanced classes. Depending on the algorithm you use and metrics you optimise for (seems you do it for the accuracy) you will get weird behaviour because of the imbalanced classes.